devito/arch/archinfo.py

```diff
@@ -649,9 +647,23 @@ def memavail(self, *args, **kwargs):
         """Available physical memory in bytes, or None if unknown."""
         return None
 
+    def max_mem_trans_size(self, dtype):
+        """
+        Number of items of type `dtype` that can be transferred in a single
+        memory transaction.
+        """
+        assert self.max_mem_trans_nbytes % np.dtype(dtype).itemsize == 0
```

Comment:
@EdCaunt 
No, because we developers write these classes and upon testing a new one we would be thrown the assert exception right away if the case

@mloubout yes, == 0, the memory transaction nbytes is always a multiple of 2, just like a C base type size